CMake/SetTargetSourceFolder.cmake

```diff
@@ -0,0 +1,19 @@
+function(set_target_source_folder target)
```

Comment:
`dolphin_figure_that_out()`?
Feel free to suggest better names, I suck at that sometimes!